src/unix/mod.rs

```diff
@@ -68,12 +68,28 @@ s! {
 
     // linux x32 compatibility
     // See https://sourceware.org/bugzilla/show_bug.cgi?id=16437
+    // On AIX, <stat.h> declares 'tv_nsec' as 'long', but the underlying
+    // system calls return a 4-byte value in both 32-bit and 64-bit modes.
+    // It is declared as 'i32' to avoid using the other undefined 4 bytes
+    // in the 64-bit mode.
     #[cfg(not(target_env = "gnu"))]
     pub struct timespec {
```

Comment:
To make things a bit cleaner, maybe exclude `target_os = "aix"` here? Then put the struct in `src/unix/aix/mod.rs`